notifications/pipeline_steps/base.py

```diff
@@ -24,5 +24,7 @@ def process_message(message):
         logging.info("%s has received %s %s", message.current_step, message.name, message.uuid)
         if pytz.utc.localize(datetime.utcnow()) > message.expiration_time:
```

Comment:
Checking for expiry at each pipeline step instead allows for dropping of the message as soon as it expires and return to the caller. If we checked before and after the call though, we wouldn't be able to drop a message that was stuck in a step for whatever reason. I think it's desirable to keep it this way.